### PR TITLE
add sink/source api reference

### PR DESCRIPTION
**Pull request by gangtao**
add api reference to available sinks and sources

### CODE REVIEW

File: docs/destination.md [hunk 1]
```diff
@@ -92,3 +92,102 @@ The Timeplus source plugin for Airbyte is in the early stage. Please contact us
 ## Trigger Actions via webhook{#webhook}
 
 You can also add automations to trigger other systems to take actions when Timeplus finds any real-time insights. Simply choose the **Webhook** as the action type and optionally set a message body (by default, the entire row will be encoded as a JSON document and sent to the webhook). You can use this approach to perform rule-based automation without human interaction, such as swapping a overheated equipment, scaling up to scaling down the server farm, or reminder users on slack, etc. Please check [this blog](https://www.timeplus.com/post/build-a-real-time-security-app-in-3-easy-steps) for real-world examples.
+
+
+## Sink API
+In case calling API to create a sink, here are the reference
```
Comment (jovezhong) [inline docs/destination.md:98]: "In the case"
add either : or .
"here are the reference" is not proper

File: docs/destination.md [hunk 2]
```diff
@@ -92,3 +92,102 @@ The Timeplus source plugin for Airbyte is in the early stage. Please contact us
...
+
+### kafka
```
Comment (jovezhong) [inline docs/destination.md:100]: might be okay to use `kafka` as the source type in the API. Ideally should be Apache Kafka

File: docs/destination.md [hunk 3]
```diff
@@ -92,3 +92,102 @@ The Timeplus source plugin for Airbyte is in the early stage. Please contact us
...
+
+refet to [https://kafka.apache.org/](https://kafka.apache.org/)
```
Comment (jovezhong) [inline docs/destination.md:102]: typo of refet

File: docs/destination.md [hunk 4]
```diff
@@ -92,3 +92,102 @@ The Timeplus source plugin for Airbyte is in the early stage. Please contact us
...
+
+| Property                | Required    | Description                                               | Default |
+|-------------------------|---------|-----------------------------------------------------------|---------|
+| brokers                   | yes | Specifies the list of broker addresses. This is a comma-separated string. such as `kafka1:9092,kafka2:9092,kafka3:9092`| | |
+| topic                     | yes | Specifies the Kafka topic to send data to.              | |
+| batch_count               | no  | Specifies the Kafka offset configuration.    support `latest,earliest`            |`1000`|
+| data_type                 | yes | Specifies the data type to use for creating the stream.   support `json`,` `,`protobuf` | |
```
Comment (jovezhong) [inline docs/destination.md:109]: the empty string ` ` is not obvious in the rendered doc

File: docs/destination.md [hunk 5]
```diff
@@ -92,3 +92,102 @@ The Timeplus source plugin for Airbyte is in the early stage. Please contact us
...
+refet to [https://kafka.apache.org/](https://kafka.apache.org/)
+
+| Property                | Required    | Description                                               | Default |
+|-------------------------|---------|-----------------------------------------------------------|---------|
+| brokers                   | yes | Specifies the list of broker addresses. This is a comma-separated string. such as `kafka1:9092,kafka2:9092,kafka3:9092`| | |
+| topic                     | yes | Specifies the Kafka topic to send data to.              | |
+| batch_count               | no  | Specifies the Kafka offset configuration.    support `latest,earliest`            |`1000`|
+| data_type                 | yes | Specifies the data type to use for creating the stream.   support `json`,` `,`protobuf` | | 
+| sasl                      | no  | Specifies the Simple Authentication and Security Layer (SASL) mechanism for authentication. support `none`,`plain`,`scram-sha-256`,`scram-sha-512` | `none` | |
+| username                  | no  | Specifies the username for authentication.               ||
```
Comment (jovezhong) [inline docs/destination.md:111]: May need to mention it is required when you connect to Confluent Cloud

File: docs/destination.md [hunk 6]
```diff
@@ -92,3 +92,102 @@ The Timeplus source plugin for Airbyte is in the early stage. Please contact us
...
+| password                  | no  | Specifies the password for authentication.               ||
+| tls.disable               | no  | If set to `true`, disables TLS encryption.               |`false`|
+| tls.skip_verify_server    | no  | If set to `true`, skips server certificate verification when using TLS. |`false`|
+| protobuf_schema           | no  | Specifies the Protobuf schema.                          |
```
Comment (jovezhong) [inline docs/destination.md:115]: May need to mention we need the full text of schema, not just a path or URI

File: docs/destination.md [hunk 7]
```diff
@@ -92,3 +92,102 @@ The Timeplus source plugin for Airbyte is in the early stage. Please contact us
...
+| protobuf_message          | no  | Specifies the Protobuf message type.                     |
+
+
+### http
+
+| Property                | Required    | Description                                               | Default |
+|-------------------------|---------|-----------------------------------------------------------|---------|
+| url                      | yes  | Specifies the url of http                      |        |
```
Comment (jovezhong) [inline docs/destination.md:123]: the URL of the HTTP source

File: docs/destination.md [hunk 8]
```diff
@@ -92,3 +92,102 @@ The Timeplus source plugin for Airbyte is in the early stage. Please contact us
...
+| protobuf_schema           | no  | Specifies the Protobuf schema.                          |
+| protobuf_message          | no  | Specifies the Protobuf message type.                     |
+
+
+### http
+
+| Property                | Required    | Description                                               | Default |
+|-------------------------|---------|-----------------------------------------------------------|---------|
+| url                      | yes  | Specifies the url of http                      |        | 
+| content_type             | no   | Specifies the content type.                    |        |
```
Comment (jovezhong) [inline docs/destination.md:124]: Some sentences end with . some not

File: docs/destination.md (hunk 1 above)
Comment (jovezhong) [reply docs/destination.md:98]: Better to leave more docs about how to call the sink API

File: docs/source.md [hunk 9]
```diff
@@ -4,3 +4,72 @@ Timeplus integrates with a wide range of systems as data sources, such as Apache
 
 You can define one or more sources to set up the background jobs to load data into Timeplus. Please check the [Data Ingestion](ingestion) section for more details.
 
+
+## Source API
+In case calling API to create a source, here are the reference
```
Comment (jovezhong) [inline docs/source.md:9]: similar issues/suggestions as sink API

File: docs/source.md [hunk 10]
```diff
@@ -4,3 +4,72 @@ Timeplus integrates with a wide range of systems as data sources, such as Apache
...
+
+### kafka
+
+refet to [https://kafka.apache.org/](https://kafka.apache.org/)
```
Comment (jovezhong) [inline docs/source.md:13]: typo

**Review by jovezhong — CHANGES_REQUESTED** [review]
Please pay attentions to the details

File: docs/destination.md (hunk 4 above)
Comment (ye11ow) [reply docs/destination.md:109]: Maybe we don't need to mention protobuf for now? Right now the protobuf is too PoC. For example, in order to use protobuf, the user need to pass `protobuf_schema` and `protobuf_message` that matches the output of Sink. However, it is very difficult for the user to write the schema by themselves. We should dynamically analyze the SQL result and create the schema for them

File: docs/destination.md [hunk 11]
```diff
@@ -92,3 +92,102 @@ The Timeplus source plugin for Airbyte is in the early stage. Please contact us
...
+| http_method              | no   | Specifies the password for authentication.     | `POST` |
+| payload_field            | no   | If set to `true`, disables TLS encryption.     |        |
```
Comment (ye11ow) [inline docs/destination.md:126]: the payload of the http request

File: docs/destination.md [hunk 12]
```diff
@@ -92,3 +92,102 @@ The Timeplus source plugin for Airbyte is in the early stage. Please contact us
...
+| scopes                  | no     | scopes, list of strings                    |        | 
+
+
+### slack
+
+refer to [https://slack.com/](https://slack.com/)
+
+| Property                | Required| Description                                               | Default |
+|-------------------------|---------|-----------------------------------------------------------|---------|
+| url                    | yes     | the webhook url of the slack channel                    |   |
```
Comment (ye11ow) [inline docs/destination.md:152]: Let's mention that it will be considered as secret in T+

File: docs/destination.md [hunk 13]
```diff
@@ -92,3 +92,102 @@ The Timeplus source plugin for Airbyte is in the early stage. Please contact us
...
+
+
+### slack
+
+refer to [https://slack.com/](https://slack.com/)
+
+| Property                | Required| Description                                               | Default |
+|-------------------------|---------|-----------------------------------------------------------|---------|
+| url                    | yes     | the webhook url of the slack channel                    |   | 
+| template               | no     | the template used to send query result to slack, use `{{ .field_name }}` to replace the field of query result you want to reference                              |        |
```
Comment (ye11ow) [inline docs/destination.md:153]: if `template` is empty, we will encode the event into JSON format

File: docs/destination.md [hunk 14]
```diff
@@ -92,3 +92,102 @@ The Timeplus source plugin for Airbyte is in the early stage. Please contact us
...
+| header                 | no     | the http header                             |   `{}`   | 
+
+
+### timeplus
+
+send query result to another timeplus stream
+
+| Property                | Required| Description                                               | Default |
+|-------------------------|---------|-----------------------------------------------------------|---------|
+| urlstream_name          | yes     | the name of the target stream |   |
```
Comment (ye11ow) [inline docs/destination.md:163]: `urlstream_name`?

File: docs/destination.md [hunk 15]
```diff
@@ -92,3 +92,102 @@ The Timeplus source plugin for Airbyte is in the early stage. Please contact us
...
+
+### kafka
+
+refet to [https://kafka.apache.org/](https://kafka.apache.org/)
+
+| Property                | Required    | Description                                               | Default |
+|-------------------------|---------|-----------------------------------------------------------|---------|
+| brokers                   | yes | Specifies the list of broker addresses. This is a comma-separated string. such as `kafka1:9092,kafka2:9092,kafka3:9092`| | |
+| topic                     | yes | Specifies the Kafka topic to send data to.              | |
+| batch_count               | no  | Specifies the Kafka offset configuration.    support `latest,earliest`            |`1000`|
```
Comment (ye11ow) [inline docs/destination.md:108]: wrong description

File: docs/source.md [hunk 16]
```diff
@@ -4,3 +4,72 @@ Timeplus integrates with a wide range of systems as data sources, such as Apache
...
+| group                     | no  | Specifies the Kafka consumer group. use the source uuid with prefix `timeplus-source-` as the default value if user does not specify it                     | `timeplus-source-<uuid>`| 
+| sasl                      | no  | Specifies the Simple Authentication and Security Layer (SASL) mechanism for authentication. support `none`,`plain`,`scram-sha-256`,`scram-sha-512` | `none` | |
+| username                  | no  | Specifies the username for authentication.               ||
+| password                  | no  | Specifies the password for authentication.               ||
+| tls.disable               | no    | If set to `true`, disables TLS encryption.               |`false`|
+| tls.skip_verify_server    | no    | If set to `true`, skips server certificate verification when using TLS. |`false`|
+| protobuf_schema           | no  | Specifies the Protobuf schema.                          |
+| protobuf_message          | no  | Specifies the Protobuf message type.                     |
+| schema_registry_address   | no  | Specifies the URL of the Schema Registry for Kafka.      |
```
Comment (ye11ow) [inline docs/source.md:29]: schema registry only applies to `avro`

File: docs/source.md [hunk 17]
```diff
@@ -4,3 +4,72 @@ Timeplus integrates with a wide range of systems as data sources, such as Apache
...
+
+
+### stream_generator
+
+a source that generates randome data for test
+
+| Property                | Required    | Description                                               | Default |
+|-------------------------|---------|-----------------------------------------------------------|---------|
+| template    | yes      | Specifies the template used to generate data, support `iot`,`user_logins`,`devops`.            |
+| interval    | yes      | Specifies the event interval. for example `200ms`       |
```
Comment (ye11ow) [inline docs/source.md:41]: `interval` is optional

File: docs/source.md [hunk 18]
```diff
@@ -4,3 +4,72 @@ Timeplus integrates with a wide range of systems as data sources, such as Apache
...
+
+| Property                | Required    | Description                                               | Default |
+|-------------------------|---------|-----------------------------------------------------------|---------|
+| topic    | yes      | Specifies the topic of the pulsar to connect to.            |
+| broker_url    | yes      | Specifies the URL of the broker to connect to.            |
+| auth_type     | yes    | Specifies the authentication type to use.  support ` `,`oauth2`,`token`           |
+| auth_params   | no | Specifies authentication parameters as key-value pairs.  | `{}` |
+
+
+### livepeer
```
Comment (ye11ow) [inline docs/source.md:56]: do we need public doc for livepeer?

**Review by ye11ow — CHANGES_REQUESTED** [review]

File: docs/destination.md (hunk 2 above)
Comment (gangtao) [reply docs/destination.md:100]: here the name is same as what we call in the api

File: docs/destination.md (hunk 6 above)
Comment (gangtao) [reply docs/destination.md:115]: as suggested by Calvin, we plan remove protobuf in the doc

File: docs/source.md (hunk 18 above)
Comment (gangtao) [reply docs/source.md:56]: as a partner, yes

**Review by jovezhong — APPROVED** [review]

**Comment by jovezhong** [comment]
(Jove Github Bot) added it to the current sprint.

**Comment by jovezhong** [comment]
(Jove Github Bot) moved this ticket out of the GitHub project(up to 1200 tickets for one project).

**Pull request merged**